lib/notes.js

```diff
@@ -0,0 +1,77 @@
+'use strict';
+const Input = require('./input.js');
+const schemaModule = require('../models/note.schema.js');
+const mongoose = require('mongoose');
+
+class NoteHandler {
+  constructor(command) {
+    if (command && command.action) {
+      this.excute(command);
+
+    }
+    else {
+      console.error('Invalid Input');
+    }
+  }
+  excute(command) {
+    console.log('excute ', command);
+    switch (command.action) {
+    case 'add':
+      this.add(command.payload, command.category);
```

Comment:
Here, you're calling the add function within this class, but no add function is defined! So this is the main source of your error. You should define an add function in this class that adds the note to the database (probably by calling the `create` function within your model